enclave/core/sgx/memory.c

```diff
@@ -9,7 +9,12 @@ bool oe_is_within_enclave(const void* p, size_t n)
     uint64_t range_start = (uint64_t)p;
     uint64_t range_end = range_start + (n == 0 ? 1 : n);
     uint64_t enclave_start = (uint64_t)__oe_get_enclave_base();
-    uint64_t enclave_end = enclave_start + __oe_get_enclave_size();
+    uint64_t enclave_end = __oe_get_enclave_size() +
```

Comment:
Instead of changing the logic here, I'm proposing:
- The enclave_size (returned by the _calculate_enclave_size and saved by the enclave) holds the original enclave size as-is  for the 0-base case
- We only adjust the enclave_size when setting up the `enclave_elrange` struct.

Reply:
Done. This would also mean the changes to enclave/core/sgx/memory.c are not needed anymore since the legal range of enclave will remain enclave->start_addr + enclave->size.